src/controllers/staff.controller.ts

```diff
@@ -1 +1,55 @@
-import { StaffModel } from "../models/staff";
+import { StaffModel, staffType } from "../models/staff";
+
+export const createStaff = async (staff: staffType) => {
+  const newStaff = new StaffModel(staff);
+  await newStaff.save();
+  return newStaff;
+};
+
+export const updateStaff = async (
+  firebaseUID: string,
+  update: { key: string; value: string },
+) => {
+  const updateObj = { $set: { [update.key]: update.value } };
+  const staff = await StaffModel.findOneAndUpdate({ firebaseUID }, updateObj);
+  if (!staff) throw new Error("Staff not found");
+};
+
+export const updateStaffActive = async (
+  firebaseUID: string,
+  update: { key: string; value: boolean },
+) => {
+  const updateObj = { $set: { [update.key]: update.value } };
+  const staff = await StaffModel.findOneAndUpdate({ firebaseUID }, updateObj);
+  if (!staff) throw new Error("Staff not found");
+};
+
+export const getAllStaff = async () => {
+  const allStaff = await StaffModel.find({});
+  if (!allStaff) throw new Error("No staff found");
+  return allStaff;
+};
+
+export const getStaffByEmail = async (email: string) => {
+  const staff = await StaffModel.findOne({ email });
+  if (!staff) throw new Error("Staff not found");
+  return staff;
+};
+
+export const getStaffByID = async (firebaseUID: string) => {
+  const staff = await StaffModel.findOne({ firebaseUID });
+  if (!staff) throw new Error("Staff not found");
+  return staff;
+};
+
+export const getStaffByActive = async (active: boolean) => {
+  const staff = await StaffModel.find({ active });
+  if (!staff) throw new Error("Staff not found");
+  return staff;
+};
+
+export const getStaffByProgram = async (programs: string[]) => {
```

Comment:
The programs parameter should be of type String instead of String[] because the method should only query for one program at a time